lib/DataObject/BlockDataMarshaller/EncryptedField.php

```diff
@@ -109,7 +111,11 @@ public function encrypt($data, $params = [])
             $fd = $params['fieldDefinition'];
             $delegateFd = $fd->getDelegate();
 
-            if (method_exists($delegateFd, 'marshalBeforeEncryption')) {
+            if ($delegateFd instanceof BeforeEncryptionMarshallerInterface || method_exists($delegateFd, 'marshalBeforeEncryption')) {
+                if (!$delegateFd instanceof BeforeEncryptionMarshallerInterface) {
+                    @trigger_error(sprintf('Usage of method_exists is deprecated since version 10.1 and will be removed in Pimcore 11.' .
+                        'Implement the %s interface instead.', BeforeEncryptionMarshallerInterface::class), E_USER_DEPRECATED);
```

Comment:
What about using [Symfony's deprecation contract](https://github.com/symfony/deprecation-contracts) for triggering deprecations in a somehow standardized way? It should already be installed with the `symfony/contracts` package.

Reply:
So that's a no? Or was it resolved automatically?

Reply:
Let me first go over the rest of the open interfaces and then I will do everything in one go.

Reply:
done